components/pill/index.jsx

```diff
@@ -0,0 +1,284 @@
+
+import React from 'react';
+import createReactClass from 'create-react-class';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { shape } from 'airbnb-prop-types';
+import { PILL } from '../../utilities/constants';
+import UtilityIcon from '../utilities/utility-icon';
+import KEYS from '../../utilities/key-code';
+
+
+const Pill = createReactClass({
+	// ### Display Name
+	displayName: PILL,
+
+	propTypes: {
+		labels: shape({
+			/**
+			 * Pill's label
+			 * _Tested with Mocha framework._
+			 */
+			label: PropTypes.string,
+			/**
+			 * Pill's title
+			 * _Tested with Mocha framework._
+			 */
+			title: PropTypes.string,
+			/**
+			 * A title to use for the remove icon.
+			 * _Tested with Mocha framework._
+			 */
+			removeTitle: PropTypes.string
+		}),
+
+		assistiveText: shape({
+			/**
+			 * Assistive text to use in the remove icon. Uses removeTitle, if not specified.
+			 * _Tested with Mocha framework._
+			 */
+			remove: PropTypes.string
+		}),
+
+		/**
+		 * A set of additional styles.
+		 * _Tested with Mocha framework._
+		 */
+		className: PropTypes.string,
+
+		/**
+		 * An href to use if the pill is shown as a link.
+		 * _Tested with Mocha framework._
+		 */
+		href: PropTypes.string,
+
+		/**
+		 * Applies the bare style to the component.
+		 * _Tested with Mocha framework._
+		 */
+		bare: PropTypes.bool,
+
+		/**
+		 * Applies the error style to the component.
+		 * _Tested with Mocha framework._
+		 */
+		hasError: PropTypes.bool,
+
+		/**
+		 * The component applies the link style if either of the href or onClick properties are set. If this behavior is not desirable,
+		 * the link property forces the component to apply (true) or remove (false) the link style, regardless of whether
+		 * the href or onClick are set.
+		 * _Tested with Mocha framework._
+		 */
+		link: PropTypes.bool,
+
+		/**
+		 * SLDSIcon component to show on the left of the pill.
+		 * _Tested with Mocha framework._
+		 */
+		icon: PropTypes.element,
+
+		/**
+		 * SLDSAvatar component to show on the left of the pill.
+		 * _Tested with Mocha framework._
+		 */
+		avatar: PropTypes.element,
+
+		/**
+		 * onClick callback executes when a user clicks on the pill or presses the Enter key.
+		 * _Tested with Mocha framework._
+		 * @callback
+		 */
+		onClick: PropTypes.func,
+
+		/**
+		 * onRemove callback executes when a user clicks on the pill's remove icon or presses the delete or the backspace keys.
+		 * _Tested with Mocha framework._
+		 * @callback
+		 */
+		onRemove: PropTypes.func,
+
+		/**
+		 * Role to use for the pill. Default role: button.
+		 * _Tested with Mocha framework._
+		 */
+		role: PropTypes.string,
+
+		/**
+		 * This is a way to specify custom contents for the pill in the case a simple text label is not enough.
+		 * _Tested with Mocha framework._
+		 */
+		children: PropTypes.node
+	},
+
+	getDefaultProps () {
+		return {
+			role: 'button',
```

Comment:
I'm ok with allowing role, but am not seeing it in the SLDS docs, and so we may want to remove the default of button.

Reply:
The pill turns `span` into an active element and I believe it's better to require a role for it. Let me know if there are certain rules or examples around that. I believe also that current eslint settings check for that.

Reply:
SLDS has overruled this in my discussions with them. Eslint is error on the default best practices rule here.

Reply:
I'm not sure it's clear to me what you are suggesting: have the default role or not?